File: Homework/JavaScript/#1 - Fundamentals/#3 - Conditional Statements/6.QuadraticEquation/QuadraticEquation.js
// Write a script that enters the coefficients a, b and c of a quadratic equation a*x2 + b*x + c = 0 and calculates and prints its real roots. Note that quadratic equations may have 0, 1 or 2 real roots.

function enterNumber() {
    var a = parseFloat(document.getElementById("a").value),
		b = parseFloat(document.getElementById("b").value),
		c = parseFloat(document.getElementById("c").value);
		d,
		xOne,
		xTwo,
		x;
		
    CalculateQuadraticEquation(a, b, c);
}

function CalculateQuadraticEquation(a, b, c) {
    d = b * b + 4 * a * c;
    document.getElementById("d").value = "Discriminant = " + d.toFixed(2);
	
    if (d > 0) {
        xOne = (-b + Math.sqrt(d)) / 2 * a;
        xTwo = (-b - Math.sqrt(d)) / 2 * a;
        document.getElementById("x-one").value = "X1 = " + xOne.toFixed(2);
        document.getElementById("x-two").value = "X2 = " + xTwo.toFixed(2);
    }
    else if (d = 0) {
        x = -b / 2 * a;
        document.getElementById("x-one").value = "X1 = X2 = " + x.toFixed(2);
    }
    else {
        document.getElementById("x-one").value = "Equation has no real roots!";
    }
}

File: Homework/C#/Part I/#4 - ConsoleInputOutput/6.QuadraticEquation/QuadraticEquation.cs
// Write a program that reads the coefficients a, b and c of a quadratic equation ax2+bx+c=0 and solves it (prints its real roots).

using System;

namespace QuadraticEquation
{
    class QuadraticEquation
    {
        static void Main()
        {
            double a;
            double b;
            double c;
            double d;
            double numOne;
            double numTwo;
            double den;
            double xOne;
            double xTwo;

            Console.Write("Please enter a number of 'a': ");
            a = double.Parse(Console.ReadLine());

            Console.Write("Please enter a number of 'b': ");
            b = double.Parse(Console.ReadLine());

            Console.Write("Please enter a number of 'c': ");
            c = double.Parse(Console.ReadLine());

            d = b * b - 4 * a * c;

            if (d > 0)
            {
                numOne = -b + Math.Sqrt(d);
                numTwo = -b - Math.Sqrt(d);
                den = 2 * a;
                xOne = numOne / den;
                xTwo = numTwo / den;
                Console.WriteLine(" a = {0} ;\n b = {1} ;\n c = {2} ;\n d = {3} ;\n X1 = {4} ;\n X2 = {5}", a, b, c, d, xOne, xTwo);
            }
            else if (d == 0)
            {
                numOne = -b;
                den = 2 * a;
                xOne = numOne / den;
                Console.WriteLine(" a = {0} ;\n b = {1} ;\n c = {2} ;\n d = 0 ;\n X1 = X2 = {3}", a, b, c, xOne);
            }
            else
            {
                Console.Write("This Equal is haven't Real Equal.");
            }
        }
    }
}
